libs/MeshKernel/src/Mesh2D.cpp

```diff
@@ -1642,164 +1623,298 @@ std::tuple<meshkernel::UInt, meshkernel::UInt> Mesh2D::IsSegmentCrossingABoundar
     return {intersectedFace, intersectedEdge};
 }
 
-std::tuple<std::vector<meshkernel::Mesh::EdgeMeshPolylineIntersection>,
-           std::vector<meshkernel::Mesh::FaceMeshPolylineIntersection>>
-Mesh2D::GetPolylineIntersections(const std::vector<Point>& polyLine)
+std::tuple<meshkernel::UInt, meshkernel::UInt> Mesh2D::GetIntersectionSeed(const std::vector<Point>& polyLine,
+                                                                           const std::vector<bool>& vistedEdges) const
 {
+    UInt crossedSegmentIndex = constants::missing::uintValue;
+    UInt crossedEdgeIndex = constants::missing::uintValue;
 
-    std::vector<EdgeMeshPolylineIntersection> edgesIntersectionsResult(GetNumEdges());
-    std::vector<FaceMeshPolylineIntersection> faceIntersectionsResult(GetNumFaces());
+    // Find starting edge and segment
+    for (UInt segmentIndex = 0; segmentIndex < polyLine.size() - 1; ++segmentIndex)
+    {
+        for (UInt edgeIndex = 0; edgeIndex < GetNumEdges(); ++edgeIndex)
+        {
+            // edge already crossed, nothing to do
+            if (vistedEdges[edgeIndex])
+            {
+                continue;
+            }
+
+            const auto [isEdgeCrossed,
+                        intersectionPoint,
+                        crossProductValue,
+                        adimensionalPolylineSegmentDistance,
+                        adimensionalEdgeDistance] = AreSegmentsCrossing(polyLine[segmentIndex],
+                                                                        polyLine[segmentIndex + 1],
+                                                                        m_nodes[m_edges[edgeIndex].first],
+                                                                        m_nodes[m_edges[edgeIndex].second],
+                                                                        false,
+                                                                        m_projection);
+            if (!isEdgeCrossed)
+            {
+                continue;
+            }
+
+            crossedSegmentIndex = segmentIndex;
+            crossedEdgeIndex = edgeIndex;
+            break;
```

Comment:
is it possible to change this to

if (isEdgeCrossed)
{
   crossedSegmentIndex = ...
   crosedEdgeIndex  = ...
   break;
}


Do you think it is easier to read?